R/trust_msoa_lad_catchments.R

```diff
@@ -0,0 +1,113 @@
+# It uses output from the scripes 'trust_types.R' and 'trust_changes.R' 
+
+# Load libraries 
+library(geographr)
+library(sf)
+library(tidyverse)
+library(httr)
+library(readxl)
+
+
+# This is the data the geapgraphr::lookup_trust_msoa is on but for this need extra columns
+query_url <- "https://app.box.com/index.php?rm=box_download_shared_file&shared_name=qh8gzpzeo1firv1ezfxx2e6c4tgtrudl&file_id=f_877983829510"
+
+# Make GET request
+GET(
+  query_url,
+  write_disk(tf <- tempfile(fileext = ".xlsx"))
+)
+
+# All admissions
+catchment_populations <- read_excel(tf, sheet = "All Admissions")
+
+# Keep only 2019 data
+catchment_proportions <-
+  catchment_populations %>%
+  filter(CatchmentYear == 2019) %>%
+  select(
+    msoa_code = msoa,
+    trust_code = TrustCode,
+    patients,
+    total_patients,
+    proportion,
+    msoa_total_catchment,
+    trust_total_catchment
+  )
+
+
+# Although there are Trust catchments in the dataset, these are meant to be the sum of the msoa_catchment across each trusts but figures do not align. 
+catchment_proportions |> 
+  group_by(trust_code) |>
+  mutate(calculated_trust_total_catchment = sum(msoa_total_catchment))
+
+# Also note the total of the patients column for an MSOA is not the same as the total_patients column, is it that some patients for a given MSOA which can't be attributed to a Trust?
+# This means that the proportion column does not sum to 100% when grouped by Trust
+catchment_proportions |> 
+  select(msoa_code, patients, total_patients) |>
+  group_by(msoa_code) |>
+  mutate(calculated_total_patients = sum(patients))
+
+# Some trust codes may need updated so aligns with 'open_trust_types.feather' ----
+open_trusts <- read_rds("data/open_trust_types.rds")
+trust_changes <- read_rds("data/trust_changes.rds")
+
+trusts_to_update <- catchment_proportions |>
+  anti_join(open_trusts) |>
+  distinct(trust_code) |>
+  pull()
+
+updated_trusts <- catchment_proportions |>
+  filter(trust_code %in% trusts_to_update) |>
+  inner_join(trust_changes, by = c("trust_code" = "old_code")) |>
+  select(-c(trust_code, date)) |>
+  rename(trust_code = new_code) |>
+  relocate(trust_code, .after = msoa_code)
+
+catchment_proportions_updated <- catchment_proportions |>
+  filter(!trust_code %in% trusts_to_update) |>
+  bind_rows(updated_trusts)|>
+  group_by(msoa_code, trust_code) 
+
+# Check through updating trust codes may be duplicated combinations of msoa_code and trust_code 
+# so need to combine rows to only distinct combination of msoa_code and trust_code 
+# the total_patients column won't change because this is the total number of patients across all trusts for a given msoa 
+# and since it is the trust code changing (not msoa) will stay the same. 
+catchment_proportions_updated_grouped <- catchment_proportions_updated |>
+  group_by(msoa_code, trust_code, total_patients) |>
+  summarise_if(is.numeric, sum) |>
```

Comment:
```suggestion
  summarise(across(where(is.numeric), sum)) |>
```
`summarise_if` has been superseded, try and use a `colwise` approach in the future: https://dplyr.tidyverse.org/articles/colwise.html